modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java

```diff
@@ -107,6 +108,12 @@ public <R> JobExecution<R> executeAsync(
             throw new IllegalArgumentException("nodes must not be empty.");
         }
 
+        for (ClusterNode node : nodes) {
+            if (topologyService.getByConsistentId(node.name()) == null) {
+                return new FailedExecution<>(new NodeNotFoundException(node.name()));
+            }
+        }
```

Comment:
I think we should not fail if at least one specified node is present. Otherwise the user has to wrap all compute API calls into a retry loop because node failure is always possible.